.github/config-schema.json

```diff
@@ -338,7 +338,14 @@
         "read_only_style": "red",
         "repo_root_format": "[$before_root_path]($style)[$repo_root]($repo_root_style)[$path]($style)[$read_only]($read_only_style) ",
         "repo_root_style": null,
-        "style": "cyan bold",
+        "style": [
+          {
+            "env": null,
+            "expected_value": null,
+            "operator": null,
+            "style": "cyan bold"
+          }
+        ],
```

Comment:
Interesting approach here. Is there a better way for this to be written though? I don't know how I feel about `style.style` for the keys here. But I also haven't taken a look at everything in this PR yet. 

I did find this PR because I was thinking of contributing this but I was considering using a `when =` or something that feels more Starship. I do think this looks great though. I haven't thought my solution through yet.

Reply:
Maybe it'd be good to have both styles? Like if it's a string it gets parsed the old way, and if it's a "list of objects?" then it could use all those. But again, mostly just thinking out loud here.

Reply:
@rogeruiz thank you for your comments! I agree the names I've chosen aren't best. I'm happy to change them if anyone has any suggestions. I just wanted it to get it reviewed quickly to verify if my implementation can be merged 😄 

>Maybe it'd be good to have both styles? Like if it's a string it gets parsed the old way, and if it's a "list of objects?" then it could use all those. But again, mostly just thinking out loud here.

It's already working that way. You can use a plain string expressions as a `style` value. Here is a test for it https://github.com/cymruu/starship/blob/eb28f7f728fcd8e594cabd4ab0ebc93a190d2c2b/src/conditional_style.rs#L181